pypesto/hierarchical/base_problem.py

```diff
@@ -193,6 +193,11 @@ def check_edatas(self, edatas: list[amici.ExpData]) -> bool:
             amici.numpy.ExpDataView(edata)["observedData"] for edata in edatas
         ]
 
+        # Mask the data using the inner problem mask. This is necessary
+        # because the inner problem is aware of only the data it uses.
+        for i in range(len(data)):
+            data[i][~self.data_mask[i]] = np.nan
+
```

Comment:
This is a bug fix. It caused an issue only if the model contained multiple non-quantitative data types (e.g. semiquantitative and relative). In that case, this check always failed, as we were not masking the data in the check in the same way it was masked in the inner problem initialization.